### PR TITLE
[@types/ember__test-helpers] Add `setupOnerror` and `resetOnerror` helpers.

### PR DESCRIPTION
**Pull request by wuron**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/emberjs/ember-test-helpers/blob/master/API.md#setuponerror>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ember__test-helpers ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ember__test-helpers@latest/))
was missing the following properties:
1. name
1. init
1. included
1. treeForAddonTestSupport
1. treeForVendor
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 781e3b06b125903f9aff86f503a52f581786055f
</p>

**Review by chriskrycho — APPROVED** [review]
Thanks!

File: types/ember__test-helpers/index.d.ts [hunk 1]
```diff
@@ -223,6 +224,13 @@ declare module '@ember/test-helpers/validate-error-handler' {
     export default function(callback?: (error: Error) => void): { isValid: boolean, message: string };
 }
 
+declare module '@ember/test-helpers/onerror' {
+    import Error from '@ember/error';
+
+    export function setupOnerror(handler: (error: Error) => void): void;
```
Comment (chriskrycho) [inline types/ember__test-helpers/index.d.ts:230]: Actually, thinking on this—is this guaranteed to be an `Error`? I don’t *think* it is?

**Review by chriskrycho — CHANGES_REQUESTED** [review]

File: types/ember__test-helpers/index.d.ts (hunk 1 above)
Comment (wuron) [reply types/ember__test-helpers/index.d.ts]: Yeah, you’re right. If you just `throw "foo"` somewhere you’ll get `"foo"` in error handler.
I’ve updated the PR.

File: types/ember__test-helpers/index.d.ts [hunk 2]
```diff
@@ -223,6 +224,11 @@ declare module '@ember/test-helpers/validate-error-handler' {
     export default function(callback?: (error: Error) => void): { isValid: boolean, message: string };
 }
 
+declare module '@ember/test-helpers/onerror' {
+    export function setupOnerror(handler: (error: any) => void): void;
```
Comment (chriskrycho) [inline types/ember__test-helpers/index.d.ts:228]: One last tweak: let’s make this `unknown` instead of `any`.

**Review by chriskrycho — CHANGES_REQUESTED** [review]

File: types/ember__test-helpers/index.d.ts (hunk 2 above)
Comment (wuron) [reply types/ember__test-helpers/index.d.ts]: Good call. Updated.

**Review by chriskrycho — APPROVED** [review]
Thank you!

**Comment by chriskrycho** [comment]
Ready to merge

**Pull request merged**